### PR TITLE
Big Changes

### PR DESCRIPTION
**Pull request by NaoUnderscore**

### CODE REVIEW

**Review by xRoier — APPROVED** [review]
Looking good!

**Review by Monaldcry7788 — APPROVED** [review]

File: Scope.Client/API/Enums/ExecutionPriority.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        /// <inheritdoc cref="Medium"/>
+        Default = Medium,
+
+        /// <summary>
+        /// Execute the mod last, after other ones.
+        /// </summary>
+        Lowest = 0,
+
+        /// <summary>
+        /// Default mod execution priority.
```
Comment (TheUltiOne) [inline Scope.Client/API/Enums/ExecutionPriority.cs:21]: > Default mod execution priority

> Default = Medium

File: Scope.Client/API/Features/Client.cs [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        /// </summary>
+        public Hardware Hardware => new();
+
+        /// <summary>
+        /// Gets the current scene.
+        /// </summary>
+        public Scene CurrentScene => SceneManager.GetActiveScene();
+
+        /// <summary>
+        /// Gets a value indicating whether or not the client is connected.
```
Comment (TheUltiOne) [inline Scope.Client/API/Features/Client.cs]: “Gets a value indicating whether or not the client is connected to a server” would be better.

File: Scope.Client/Events/EventArgs/Client/LoadingMainMenuEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Scope.Client.Events.EventArgs
+{
+    /// <summary>
+    /// Contains all informations before loading main menu.
+    /// </summary>
+    public class LoadingMainMenuEventArgs : System.EventArgs
```
Comment (TheUltiOne) [inline Scope.Client/Events/EventArgs/Client/LoadingMainMenuEventArgs.cs:13]: Not related to review but can someone add it to the todo list

File: Scope.Client/Loader/Features/Configs/CommentGatheringTypeInspector.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+namespace Scope.Client.Loader.Features.Configs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using YamlDotNet.Serialization;
+    using YamlDotNet.Serialization.TypeInspectors;
+
+    /// <summary>
+    /// Spurce: https://dotnetfiddle.net/8M6iIE.
```
Comment (TheUltiOne) [inline Scope.Client/Loader/Features/Configs/CommentGatheringTypeInspector.cs]: Source*

File: Scope.Client/Loader/Features/Configs/ConfigsValidator.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Scope.Client.Loader.Features.Configs
+{
+    using System;
+    using System.ComponentModel.DataAnnotations;
+    using YamlDotNet.Core;
+    using YamlDotNet.Serialization;
+
+    /// <summary>
+    /// Configs validator.
```
Comment (TheUltiOne) [inline Scope.Client/Loader/Features/Configs/ConfigsValidator.cs]: Make this more clear like, “A class used to validate the configs in ..config path..”

File: Scope.Client/Loader/Loader.cs [hunk 6]
```diff
@@ -65,21 +157,21 @@ public static void LoadAll()
...
             foreach (IMod<IConfig> mod in toLoad)
             {
                 try
                 {
-                    if (mod.Name.StartsWith("Scope") && mod.Config.IsEnabled)
-                    {
-                        mod.OnEnabled();
-                        toLoad.Remove(mod);
-                    }
+                    if (!mod.Name.StartsWith("Scope") || !mod.Config.IsEnabled)
```
Comment (TheUltiOne) [inline Scope.Client/Loader/Loader.cs:171]: Why not load mods that names don’t start with Scope? Unless if it’s for compatibility or something

**Review by TheUltiOne — CHANGES_REQUESTED** [review]
Typos and stuff.
The rest looks great, thank you Nao ❤️

**Comment by xRoier** [comment]
We need to fix dependencies, they load but they don't at the same time, see [this](https://github.com/Scope-SL/Scope/commit/fc4ea23882615437b109da68d60fc24a7a487717)

File: Scope.Client/Loader/Loader.cs (hunk 6 above)
Comment (NaoUnderscore) [reply Scope.Client/Loader/Loader.cs:171]: It's not how this logic works, don't worry, it's fine.

**Review by TheUltiOne — APPROVED** [review]

**Review by Monaldcry7788 — APPROVED** [review]

**Pull request merged**